addons/main/functions/fnc_setUnconscious.sqf

```diff
@@ -41,6 +41,21 @@ if (_set) then {
                 };
             };
         }, [_unit, cba_missionTime], _restBleedout] call CBA_fnc_waitAndExecute;
+        [_unit, cba_missionTime, _restBleedout] spawn {
+            params ["_unit", "_downedTime", "_time"];
+            private _unconscious = (lifeState _unit) == "INCAPACITATED";
+            // cancel the while loop if they're revived so that only one timer can be going at once
+            while {_time > 0 && _unconscious && (_unit getVariable [QGVAR(bleedoutTime), -1]) isEqualTo _downedTime} do {
+                _time = _time - 1;
+                _unit setVariable [QGVAR(bleedoutTimeLeft), _time];
+                if (isMultiplayer) then
+                {
+                    uiSleep 1;
+                } else {
+                    sleep 1;
+                };
+            };
+        };
```

Comment:
This will not work in MP, as noting gets synced over to other machines.

Instead of setting each second how much time is left, which is a no go in MP optimization, you can set the time when they will die in accordance to CBA_missionTime (https://github.com/CBATeam/CBA_A3/wiki/Global-Variables). This will not be 100% accurate but it's the best we've got.

you also check `_unconscious` in the while loop. which will is a static now in this case. You will need to check `(lifeState _unit) == "INCAPACITATED"` to get the current state.